lib/phoenix_ecto/plug.ex

```diff
@@ -1,15 +1,14 @@
-defimpl Plug.Exception, for: Ecto.CastError do
-  def status(_), do: 400
-end
-
-defimpl Plug.Exception, for: Ecto.Query.CastError do
-  def status(_), do: 400
-end
-
-defimpl Plug.Exception, for: Ecto.NoResultsError do
-  def status(_), do: 404
-end
+errors = [
+  {Ecto.CastError, 400},
+  {Ecto.Query.CastError, 400},
+  {Ecto.NoResultsError, 404},
+  {Ecto.InvalidChangesetError, 422},
+]
 
-defimpl Plug.Exception, for: Ecto.InvalidChangesetError do
-  def status(_), do: 422
+for {exception, status_code} <- errors do
+  unless exception in Application.get_env(:phoenix_ecto, :exclude_ecto_excaptions, []) do
```

Comment:
Also, the default value of the environment variable should be set inside `def application` in the mix.exs file:

    def application do
      [env: [exclude_ecto_exceptions_from_plug: []]
    end